### ISSUE TITLE
init_by_lua runs on every request with lua_code_cache off?

### ISSUE THREAD

**Issue by alexyuriev** [openresty/openresty#519]
Hi there,

It seems that with lua_code_cache off every single request that lands on nginx would trigger execution of code in init_by_lua/init_by_lua_file/etc. This seems to be an odd choice as based on the state machine init_by_lua should only run during nginx initialization and it does not run per request should the lua_code_cache be on.

Is it intentional? If so, is there a way to check inside the lua code if the code cache is on so be able to bypass the block?

**Comment by agentzh** [comment]
@alexyuriev Yes, this is intentional because every request has to have its own LuaJIT VM instance in the `lua_code_cache off` mode due to some limitations in LuaJIT.

**Comment by alexyuriev** [comment]
@agentzh Sorry, I do not understand.

if init_by_lua runs in the init stage of nginx ( i.e. at the config processing stage ) as the diagram shows it, then why would it have any bearing to the per-request VM?

Or is the diagram wrong?
